### PR TITLE
strapi to mdx converted beta

### PR DESCRIPTION
**Pull request by modernpacifist**

### CODE REVIEW

File: strapi_to_mdx_converter.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+                f.write(f"<Carousel effect={component.get('switchType')}>\n")
+                for carousel_photo in component.get('media'):
+                    f.write(f"\t<Image src={carousel_photo.get('url')} alt=hehe />\n")
+                f.write(f"</Carousel>\n")
+
+
+def main():
+    paths = rq.get(PATHS_URL).text
+
+    json_paths = json.loads(paths)
```
Comment (tmshv) [inline strapi_to_mdx_converter.py:70]: use `.json()`

File: strapi_to_mdx_converter.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+            if component.get("__component") == "unit-4.image":
+                f.write(f"<Image src={component.get('image').get('url')} alt=hehe />\n")
+
+            if component.get("__component") == "unit-4.carousel":
+                f.write(f"<Carousel effect={component.get('switchType')}>\n")
+                for carousel_photo in component.get('media'):
+                    f.write(f"\t<Image src={carousel_photo.get('url')} alt=hehe />\n")
+                f.write(f"</Carousel>\n")
+
+    with open(f"./data/{slug_filename}.en.mdx", "w") as f:
```
Comment (tmshv) [inline strapi_to_mdx_converter.py:49]: what the hell?
remove this duplication

File: strapi_to_mdx_converter.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    if not os.path.exists(directory):
+        os.makedirs(directory)
+
+
+def get_all_pages(json_list):
+    return [i.get('slug') for i in json_list]
+
+
+def get_page_contents(slug):
+    return json.loads(rq.get(f"{CONTENT_PAGE_URL}{slug}").text)
```
Comment (tmshv) [inline strapi_to_mdx_converter.py:23]: this function do nothing

File: strapi_to_mdx_converter.py [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+    excerpt = json_contents.get("excerpt")
+
+    with open(f"./data/{slug_filename}.ru.mdx", "w") as f:
+        f.write(f"---\ncover: {slug_filename}\nexcerpt: {excerpt}\n---\n\n")
+        f.write(f"# {json_contents.get('title')}\n")
+
+        for component in json_contents.get("content"):
+            if component.get("__component") == "unit-4.text":
+                f.write(f"{component.get('text')}\n")
```
Comment (tmshv) [inline strapi_to_mdx_converter.py:38]: add empty line after each component

File: strapi_to_mdx_converter.py [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+            if component.get("__component") == "unit-4.image":
+                f.write(f"<Image src={component.get('image').get('url')} alt=hehe />\n")
```
Comment (tmshv) [inline strapi_to_mdx_converter.py:41]: you should get `src="value"`
quotes!

**Review by tmshv — CHANGES_REQUESTED** [review]

File: strapi_to_mdx_converter.py [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+                f.write(f"</Carousel>\n\n")
+
+
+def main():
+    paths = rq.get(PATHS_URL).json()
+    all_pages = get_all_pages(paths)
+
+    for page in tqdm(all_pages):
+        slug_filename = page.replace("/", "")
+        json_page_data = json.loads(rq.get(f"{CONTENT_PAGE_URL}{page}").text)
```
Comment (tmshv) [inline strapi_to_mdx_converter.py:61]: wtf?

File: strapi_to_mdx_converter.py [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+
+            if component.get("__component") == "unit-4.carousel":
+                f.write(f"<Carousel effect={component.get('switchType')}>\n")
+                for carousel_photo in component.get('media'):
+                    f.write(f"\t<Image src=\"{carousel_photo.get('url')}\" alt=\"{carousel_photo.get('alternativeText')}\" />\n")
+                f.write(f"</Carousel>\n\n")
+
+
+def main():
+    paths = rq.get(PATHS_URL).json()
```
Comment (tmshv) [inline strapi_to_mdx_converter.py:56]: need to check response is ok before getting json

File: strapi_to_mdx_converter.py [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+def save_dir_init(directory):
+    if not os.path.exists(directory):
+        os.makedirs(directory)
+
+
+def get_all_pages(json_list):
+    return [i.get('slug') for i in json_list]
+
+
+def save_mdx_files(slug_filename, json_data, lang_flag=False):
```
Comment (tmshv) [inline strapi_to_mdx_converter.py:22]: this is not a lang flag as i see
this is `en_flag`

File: strapi_to_mdx_converter.py [hunk 9]
```diff
@@ -0,0 +1,67 @@
+#!/bin/env python3.9
+import requests as rq
+import json
+import os
+
+from tqdm import tqdm
+
+
+PATHS_URL = "https://unit.tmshv.com/unit-4-pages/paths"
+CONTENT_PAGE_URL = "https://unit.tmshv.com/unit-4-pages?slug="
```
Comment (tmshv) [inline strapi_to_mdx_converter.py:10]: `requests` can handle query much better than this

File: strapi_to_mdx_converter.py [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+
+
+def save_dir_init(directory):
+    if not os.path.exists(directory):
+        os.makedirs(directory)
+
+
+def get_all_pages(json_list):
+    return [i.get('slug') for i in json_list]
```
Comment (tmshv) [inline strapi_to_mdx_converter.py:19]: this function do nothing

**Review by tmshv — CHANGES_REQUESTED** [review]

File: strapi_to_mdx_converter.py (hunk 6 above)
Comment (modernpacifist) [reply strapi_to_mdx_converter.py:61]: Can you be more specific

File: strapi_to_mdx_converter.py [hunk 11]
```diff
@@ -53,14 +48,21 @@ def save_mdx_files(slug_filename, json_data, lang_flag=False):
 
 
 def main():
-    paths = rq.get(PATHS_URL).json()
-    all_pages = get_all_pages(paths)
+    if rq.get(PATHS_URL).ok:
+        paths = rq.get(PATHS_URL).json()
```
Comment (tmshv) [inline strapi_to_mdx_converter.py:52]: wtf man? are you kidding me?

File: strapi_to_mdx_converter.py [hunk 12]
```diff
@@ -53,14 +48,21 @@ def save_mdx_files(slug_filename, json_data, lang_flag=False):
...
 
     for page in tqdm(all_pages):
-        slug_filename = page.replace("/", "")
-        json_page_data = json.loads(rq.get(f"{CONTENT_PAGE_URL}{page}").text)
-        save_mdx_files(slug_filename, json_page_data)
-        save_mdx_files(slug_filename, json_page_data, True)
+        slug = page.replace("/", "")
+
+        try:
+            json_page_data = rq.get(f"{CONTENT_PAGE_URL}/{slug}").json()
```
Comment (tmshv) [inline strapi_to_mdx_converter.py:60]: what are you trying to reach by removing slash at 57 and adding it here?

**Review by tmshv — CHANGES_REQUESTED** [review]

File: strapi_to_mdx_converter.py (hunk 12 above)
Comment (modernpacifist) [reply strapi_to_mdx_converter.py:60]: removed variable that I use only once

File: strapi_to_mdx_converter.py [hunk 13]
```diff
@@ -0,0 +1,71 @@
...
+                for carousel_photo in component.get('media'):
+                    f.write(f"\t<Image src=\"{carousel_photo.get('url')}\" alt=\"{carousel_photo.get('alternativeText')}\" />\n")
+                f.write(f"</Carousel>\n\n")
+
+
+def main():
+    resp = rq.get(PATHS_URL)
+
+    if resp.ok:
+        paths = resp.json()
```
Comment (tmshv) [inline strapi_to_mdx_converter.py:54]: you think it is ok?

**Review by tmshv — CHANGES_REQUESTED** [review]

File: strapi_to_mdx_converter.py (hunk 13 above)
Comment (modernpacifist) [reply strapi_to_mdx_converter.py:54]: at least now its a single request

Comment (tmshv) [reply strapi_to_mdx_converter.py:54]: and doesn't matter it can potentially fail, right?

**Review by tmshv — CHANGES_REQUESTED** [review]
still not enough
`requests` will raise exception if you are offline
but successful request might be not ok (http code is not 2XX)

File: strapi_to_mdx_converter.py [hunk 14]
```diff
@@ -0,0 +1,79 @@
...
+                json_page_data = rq.get(f"{CONTENT_PAGE_URL}/{slug}").json()
+            except Exception as e:
+                print(e)
+                continue
+
+            save_mdx_files(slug, json_page_data)
+            save_mdx_files(slug, json_page_data, True)
+
+    except KeyboardInterrupt:
+        exit()
```
Comment (tmshv) [inline strapi_to_mdx_converter.py:75]: will not be catched on `ctrl+c` in loading pages state

File: strapi_to_mdx_converter.py [hunk 15]
```diff
@@ -0,0 +1,79 @@
...
+                f.write(f"</Carousel>\n\n")
+
+
+def main():
+    try:
+        resp = rq.get(PATHS_URL)
+    except Exception as e:
+        exit(e)
+
+    if resp.status_code != 200:
```
Comment (tmshv) [inline strapi_to_mdx_converter.py:56]: explain WHY not ok?

File: strapi_to_mdx_converter.py [hunk 16]
```diff
@@ -0,0 +1,79 @@
...
+        exit()
```
Comment (tmshv) [inline strapi_to_mdx_converter.py:57]: 1

**Review by tmshv — CHANGES_REQUESTED** [review]

File: strapi_to_mdx_converter.py (hunk 15 above)
Comment (modernpacifist) [reply strapi_to_mdx_converter.py:56]: response.ok returns True if status_code is less than 400, otherwise False.

Comment (tmshv) [reply strapi_to_mdx_converter.py:56]: that's mean it is ok

File: strapi_to_mdx_converter.py [hunk 17]
```diff
@@ -0,0 +1,79 @@
...
+        resp = rq.get(PATHS_URL)
+    except Exception as e:
+        exit(e)
+
+    if resp.status_code != 200:
+        exit(1)
+
+    try:
+        all_pages = [i.get('slug') for i in resp.json()]
+
```
Comment (tmshv) [inline strapi_to_mdx_converter.py:61]: bug still here

**Review by tmshv — CHANGES_REQUESTED** [review]

**Review by tmshv — CHANGES_REQUESTED** [review]
much better 
wanted to rewrite generation code template engine like handlebars?

**Review by tmshv — APPROVED** [review]

**Pull request merged**